File: specification/connectedvmware/resource-manager/Microsoft.ConnectedVMwarevSphere/preview/2022-01-10-preview/examples-java/CreateVCenter.java
import com.azure.resourcemanager.connectedvmware.models.ExtendedLocation;
import com.azure.resourcemanager.connectedvmware.models.VICredential;

/** Samples for VCenters Create. */
public final class Main {
    /*
     * x-ms-original-file: specification/connectedvmware/resource-manager/Microsoft.ConnectedVMwarevSphere/preview/2022-01-10-preview/examples/CreateVCenter.json
     */
    /**
     * Sample code: CreateVCenter.
     *
     * @param manager Entry point to ConnectedVMwareManager.
     */
    public static void createVCenter(com.azure.resourcemanager.connectedvmware.ConnectedVMwareManager manager) {
        manager
            .vCenters()
            .define("ContosoVCenter")
            .withRegion("East US")
            .withExistingResourceGroup("testrg")
            .withFqdn("ContosoVMware.contoso.com")
            .withExtendedLocation(
                new ExtendedLocation()
                    .withType("customLocation")
                    .withName(
                        "/subscriptions/a5015e1c-867f-4533-8541-85cd470d0cfb/resourceGroups/demoRG/providers/Microsoft.ExtendedLocation/customLocations/contoso"))
            .withPort(1234)
            .withCredentials(new VICredential().withUsername("tempuser").withPassword("<password>"))
            .create();
    }
}

File: specification/connectedvmware/resource-manager/Microsoft.ConnectedVMwarevSphere/preview/2022-01-10-preview/examples-js/CreateVCenter.js
const { AzureArcVMwareManagementServiceAPI } = require("@azure/arm-connectedvmware");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Create Or Update vCenter.
 *
 * @summary Create Or Update vCenter.
 * x-ms-original-file: specification/connectedvmware/resource-manager/Microsoft.ConnectedVMwarevSphere/preview/2022-01-10-preview/examples/CreateVCenter.json
 */
async function createVCenter() {
  const subscriptionId = "fd3c3665-1729-4b7b-9a38-238e83b0f98b";
  const resourceGroupName = "testrg";
  const vcenterName = "ContosoVCenter";
  const body = {
    credentials: { password: "<password>", username: "tempuser" },
    extendedLocation: {
      name: "/subscriptions/a5015e1c-867f-4533-8541-85cd470d0cfb/resourceGroups/demoRG/providers/Microsoft.ExtendedLocation/customLocations/contoso",
      type: "customLocation",
    },
    fqdn: "ContosoVMware.contoso.com",
    location: "East US",
    port: 1234,
  };
  const options = { body };
  const credential = new DefaultAzureCredential();
  const client = new AzureArcVMwareManagementServiceAPI(credential, subscriptionId);
  const result = await client.vCenters.beginCreateAndWait(resourceGroupName, vcenterName, options);
  console.log(result);
}

createVCenter().catch(console.error);
